BUG: Build failure with Xcode 12

<!-- 
Thank you for taking the time to file a bug report. 
Please fill in the fields below, deleting the sections that 
don't apply to your issue. You can view the final output
by clicking the preview button above.
Note: This is a comment, and won't appear in the output.
-->

Xcode 12 for OS X was just released, and I can't build scipy with it. The command `pip3 install . -v --user` fails, ending with
```
  error: Command "clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/usr/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk/System/Library/Frameworks/Tk.framework/Versions/8.5/Headers -DUSE_VENDOR_BLAS=1 -Iscipy/sparse/linalg/dsolve/SuperLU/SRC -I/private/var/folders/cp/n8wtqs490tq5psknff1hv9qr0000gn/T/pip-build-env-k4q2ai5g/overlay/lib/python3.8/site-packages/numpy/core/include -c scipy/sparse/linalg/dsolve/SuperLU/SRC/clacon2.c -o build/temp.macosx-10.15-x86_64-3.8/scipy/sparse/linalg/dsolve/SuperLU/SRC/clacon2.o -MMD -MF build/temp.macosx-10.15-x86_64-3.8/scipy/sparse/linalg/dsolve/SuperLU/SRC/clacon2.o.d" failed with exit status 1
  Building wheel for scipy (PEP 517): finished with status 'error'
  ERROR: Failed building wheel for scipy
Failed to build scipy
ERROR: Could not build wheels for scipy which use PEP 517 and cannot be installed directly
```

This may be the same issue as https://github.com/scipy/scipy/issues/12656 — at least the error at the end is the same — but that issue was closed while only providing the suggestion to use a different compiler. 

I am using Xcode 12, plus homebrew's Python 3.8, pip3, and gfortran. Unfortunately, I am really trying to figure out how to get this to build in Sage (sagemath.org), and using a gcc toolchain instead of clang would be, at the least, very inconvenient.

Yes, I can confirm this, too (i.e. both the stable Xcode 12.0 and Xcode 12.2.beta are not. Apple's versioning is far from perfect. Perhaps the easiest identifucation of the toolchain is from the C compiler (clang) version
```
Apple clang version 12.0.0 (clang-1200.0.32.4)
```
note that `xcode-select -v ` still outputs version 2373 - so it was not bumped.

A typical error comes from a missing `extern` declaration, such as 
```
scipy/sparse/linalg/dsolve/SuperLU/SRC/zlacon2.c:175:5: error: implicit declaration of function 'zcopy_' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
      zcopy_(n, x, &c__1, v, &c__1);
      ^
```

I'll try to fix them all and make a PR

--- a/scipy/spatial/qhull_misc.c
+++ b/scipy/spatial/qhull_misc.c
@@ -1,4 +1,4 @@
-#include "qhull_src/src/libqhull_r.h"
+#include "qhull_src/src/qhull_ra.h"
 
 #define trace1(args) {}
 

